aten/src/ATen/Context.cpp

```diff
@@ -96,7 +96,7 @@ bool Context::setFlushDenormal(bool on) {
 
 TypeExtendedInterface& getType(TensorOptions options) {
   return globalContext().getType(
-            options.backend(), options.dtype(), options.is_variable());
+            options.backend(), dataTypeToScalarType(options.dtype().id()), options.is_variable());
```

Comment:
Now that `TypeMeta` is just a pointer, there isn't really any reason for `dataTypeToScalarType` to take an ID; just make it take the actual `TypeMeta`.